packages/base/src/signals/signal-manager.ts

```diff
@@ -141,6 +143,9 @@ export class SignalManager extends EventEmitter implements SignalManager {
     fireOverviewOutputSelectedSignal(payload: { traceId: string, outputDescriptor: OutputDescriptor}): void {
         this.emit(Signals.OVERVIEW_OUTPUT_SELECTED, payload);
     }
+    fireNewActiveUnitController(unitController: TimeGraphUnitController): void {
+        this.emit(Signals.NEW_ACTIVE_UNIT_CONTROLLER, unitController);
```

Comment:
passing the unitcontroller instance on to the time range widget will work in this application. However, it will not work when integrating it in the vscode-trace-extension, because the time range widget and the trace viewer are running in webviews where each webview can't share objects. So, you will have to implement this feature a bit differently and send the selection start/end and view range start/end  in new signal.

Besides, you will have to manage when experiments are switched. Not sure if you can use the existing experiment selected signal for that.